### PR TITLE
Redis Token Store

### PR DESCRIPTION
**Pull request by efenderbosch**
For #367.

Has a customizable serialization strategy that defaults to strings and JDK object serialization.

### CODE REVIEW

**Comment by efenderbosch** [comment]
I've fixed the failing unit test. How do I resubmit?

**Comment by efenderbosch** [comment]
ResourceServerConfigurationTests all pass locally for me.

**Comment by dsyer** [comment]
Thanks for this. I had a merge conflict and 2 failed tests. Any chance you could rebase onto master and re-push?

**Comment by efenderbosch** [comment]
I did a git pull from master.

Here's my reactor build:

[INFO] Reactor Summary:
[INFO] 
[INFO] OAuth for Spring Security .......................... SUCCESS [ 17.167 s]
[INFO] OAuth 1(a) for Spring Security ..................... SUCCESS [ 11.943 s]
[INFO] OAuth2 for Spring Security ......................... SUCCESS [ 37.667 s]
[INFO] spring-oauth2-tests ................................ SUCCESS [  0.000 s]
[INFO] spring-oauth2-tests-common ......................... SUCCESS [  1.759 s]
[INFO] spring-oauth2-tests-vanilla ........................ SUCCESS [ 10.329 s]
[INFO] spring-oauth2-tests-mappings ....................... SUCCESS [  8.340 s]
[INFO] spring-oauth2-tests-form ........................... SUCCESS [ 11.275 s]
[INFO] spring-oauth2-tests-jwt ............................ SUCCESS [  7.781 s]
[INFO] spring-oauth2-tests-approval ....................... SUCCESS [  6.892 s]
[INFO] spring-oauth2-tests-jdbc ........................... SUCCESS [  9.443 s]
[INFO] spring-oauth-tests-custom-grant .................... SUCCESS [  7.797 s]
[INFO] spring-oauth2-tests-multi .......................... SUCCESS [  7.401 s]
[INFO] spring-oauth2-tests-client ......................... SUCCESS [  6.817 s]
[INFO] spring-oauth2-tests-resource ....................... SUCCESS [  6.167 s]
[INFO] spring-oauth2-tests-xml ............................ SUCCESS [  0.000 s]
[INFO] spring-oauth2-tests-xml-common ..................... SUCCESS [  0.684 s]
[INFO] spring-oauth2-tests-xml-vanilla .................... SUCCESS [  9.362 s]
[INFO] spring-oauth2-tests-xml-mappings ................... SUCCESS [  7.022 s]
[INFO] spring-oauth2-tests-xml-form ....................... SUCCESS [  6.955 s]
[INFO] spring-oauth2-tests-xml-jwt ........................ SUCCESS [  7.340 s]
[INFO] spring-oauth2-tests-xml-approval ................... SUCCESS [  6.874 s]
[INFO] spring-oauth2-tests-xml-jdbc ....................... SUCCESS [  7.883 s]
[INFO] spring-oauth2-tests-xml-client ..................... SUCCESS [  9.203 s]
[INFO] OAuth for Spring Security - Integration Tests ...... SUCCESS [  0.127 s]
[INFO] OAuth for Spring Security - Sparklr (OAuth 1 Provider Example) SUCCESS [  4.224 s]
[INFO] OAuth for Spring Security - Tonr (OAuth 1 Consumer Example) SUCCESS [  3.789 s]
[INFO] OAuth for Spring Security - Sparklr2 (OAuth 2 Provider Example) FAILURE [  1.201 s]
[INFO] OAuth for Spring Security - Tonr2 (OAuth 2 Client Example) SKIPPED
[INFO] OAuth for Spring Security - Samples ................ SKIPPED

**Comment by dsyer** [comment]
So that's *your* master branch? Not the one in github?

**Comment by efenderbosch** [comment]
It is master of my fork, which is up to date w/ master of parent.

What's in Sparklr2? Could my changes have impacted those tests?

**Comment by dsyer** [comment]
I doubt it. Let me try again.

**Comment by dsyer** [comment]
You must be looking at different code than me. When I rebase to master I get a conflict in `TokenStoreBaseTests` and I can't figure out what you wanted to change. Can you look again?

**Comment by efenderbosch** [comment]
Ahhh... I see now. I'll fix them shortly.

**Comment by efenderbosch** [comment]
Ok, I've done that, but the diff is empty. My apologies, I haven't much in the way of forks and pull requests before.

I changed ``TokenStoreBaseTests.testStoreRefreshToken`` because it assumes that the backing ``TokenStore`` always uses an ``ExpiringOAuth2RefreshToken``. The ``RedisTokenStore`` checks to see if the refresh token does expire and sets the TTL on the key in Redis if it is.

Also, it seems like ``ResourceServerConfigurationTests`` sometimes passes and sometimes fails.

**Comment by dsyer** [comment]
I still get a conflict. Did you "git checkout master && git fetch && git rebase origin/master"? If you do that and fix the conflict you can "git add -A . && git rebase --continue" then "git push -f" to push the new changes to the PR.

**Comment by efenderbosch** [comment]
Better now?
